### PR TITLE
 Wallet: Add Settings page

### PR DESCRIPTION
**Pull request by figitaki**
Screen Shot 2019-05-16 at 4 38 40 PM

Adds a Settings page which opens when the user clicks on the "Settings" button in the Header. Uses router, and removes the ActiveWallet context in favor of using routes.

### CODE REVIEW

File: frontend/wallet/src/main/AppWindow.re [hunk 1]
```diff
@@ -75,6 +75,11 @@ include Single.Make({
         },
       );
 
+      Bindings.Fs.watchFile(
```
Comment (bkase) [inline frontend/wallet/src/main/AppWindow.re]: nit: Can you stick a comment here about how this is to enable hot reloading

File: frontend/wallet/src/render/Hooks.re [hunk 2]
```diff
@@ -20,3 +20,11 @@ let useRoute = () => {
 
   path;
 };
+
+let useActiveWallet = () => {
```
Comment (bkase) [inline frontend/wallet/src/render/Hooks.re]: nice

File: frontend/wallet/src/render/components/TransactionCell.re [hunk 3]
```diff
@@ -331,15 +331,10 @@ let make = (~transaction: Transaction.t, ~myWallets: list(PublicKey.t)) => {
...
       <ActorName value={viewModel.sender} />
       <div className=Css.(style([marginTop(`rem(0.25))]))>
-        {ReasonReact.string(
-           switch (viewModel.action) {
-           | Transfer => " -> "
-           | Pending => " ... "
-           | Failed => " x "
-           },
-         )}
+        <span className=Theme.Text.Body.regular>{React.string("To: ")}</span>
```
Comment (bkase) [inline frontend/wallet/src/render/components/TransactionCell.re]: Are we doing this officially?

File: frontend/wallet/src/render/components/WalletItem.re [hunk 4]
```diff
@@ -46,14 +44,9 @@ module Styles = {
...
-  let (activeWallet, setActiveWallet) =
-    React.useContext(ActiveWalletProvider.context);
+  let url = ReasonReact.Router.useUrl();
 
-  let isActive =
-    Option.withDefault(
-      ~default=false,
-      Option.map(~f=active => active == wallet.key, activeWallet),
-    );
+  let isActive = url.path == ["wallet", PublicKey.toString(wallet.key)];
```
Comment (bkase) [inline frontend/wallet/src/render/components/WalletItem.re]: can you do `useWallet()` and make a `PublicKey.equal` instead for clarity?

File: frontend/wallet/src/main/AppWindow.re (hunk 1 above)
Comment (figitaki) [reply frontend/wallet/src/main/AppWindow.re]: Done.

**Pull request merged**